### ISSUE TITLE
Structs for memory mapped core register definitions

### ISSUE THREAD

**Issue by tudortimi** [ARM-software/CMSIS_6#194]
Are there any plans to also define structs for the memory mapped core registers? Currently, only bitmasking is supported. Structs would make code easier to read (at the cost of more instructions, I guess).

For example, the SAU registers only have mask/pos defines:

![image](https://github.com/user-attachments/assets/5be9784c-22a5-4894-8174-9d1b2e113f9a)

For the non-memory mapped SFRs, there are also structs (e.g. APSR) in addition to the mask/pos defines:

![image](https://github.com/user-attachments/assets/b31568d1-9a67-4121-8129-bcde70a4d222)

**Comment by JonatanAntoni** [comment]
Hi @tudortimi,

We do not have immediate plans to enhance CMSIS definitions.
Please feel free to raise a suggestion as a Pull-request.

I am happy to accept your contributions.

Thanks,
Jonatan

**Comment by JonatanAntoni** [comment]
No, we are not using any tools. The CMSIS core header files are manually maintained so far.

**Comment by tudortimi** [comment]
I'll raise a PR to add this fro one of the cores (e.g. CM33) first. It's gonna take some time, though, do you mind that the issue stays open? Also, could you assign it to me so it shows up in my TODO list?
